tests/accessor/local_accessor_api_common.h

```diff
@@ -0,0 +1,134 @@
+                          sycl::access::decorated::yes>>);
+  STATIC_CHECK(
+      std::is_same_v<
+          typename AccT::template accessor_ptr<sycl::access::decorated::no>,
+          sycl::multi_ptr<typename AccT::value_type,
+                          sycl::access::address_space::local_space,
+                          sycl::access::decorated::no>>);
+}
+
+template <typename T, typename DimensionTypeT>
+class run_api_tests {
+  static constexpr int dims = DimensionTypeT::value;
+  using AccT = sycl::local_accessor<T, dims>;
+
+ public:
+  void operator()(const std::string &type_name) {
+    auto queue = util::get_cts_object::queue();
+
+    SECTION(
+        get_section_name<dims>(type_name, "Check local_accessor alias types")) {
+      test_local_accessor_types<T, AccT, AccessModeT, TargetT>();
+    }
+
+    SECTION(get_section_name<dims>(type_name,
+                                   "Check api for empty local_accessor")) {
+      queue
+          .submit([&](sycl::handler &cgh) {
+            AccT acc;
+            test_accessor_methods_common(acc, 0 /* expected_byte_size*/,
+                                         0 /*expected_size*/,
+                                         util::get_cts_object::range<dims>::get(
+                                             0, 0, 0) /*expected_range*/);
+          })
+          .wait_and_throw();
+    }
+
+    SECTION(get_section_name<dims>(type_name, access_mode_name, target_name,
+                                   "Check api for local_accessor")) {
+      T data(expected_val);
+      constexpr size_t global_range_size = 64;
+      constexpr size_t local_range_size = 2;
+      auto global_range = util::get_cts_object::range<dims>::get(
+          global_range_size, global_range_size, global_range_size);
+      auto local_range = util::get_cts_object::range<dims>::get(
+          local_range_size, local_range_size, local_range_size);
+      sycl::nd_range<dim> nd_range(global_range, local_range);
+      bool res = false;
+      {
+        sycl::buffer<T, dims> data_buf(&data, r);
+        sycl::buffer res_buf(&res, sycl::range(1));
+        queue
+            .submit([&](sycl::handler &cgh) {
+              AccT acc(local_range, cgh);
+              AccT acc_other(local_range, cgh);
+
+              test_accessor_methods_common(
+                  acc, sizeof(T) * local_range.size() /* expected_byte_size*/,
+                  local_range.size() /*expected_size*/,
+                  local_range /*expected_range*/);
+
+              sycl::accessor res_acc(res_buf, cgh);
+              cgh.parallel_for(nd_range, [=](sycl::nd_item<dim> item) {
+                auto ref_1 = acc[0];
+
+                auto id = util::get_cts_object::id<dims>::get(1, 1, 1);
+                auto ref_2 = get_subscript_overload<T, AccT, dims>(acc, id);
+
+                res_acc[0] =
+                    std::is_same_v<decltype(ref_1), typename AccT::reference>;
```

Comment:
Don't we have multiple work-items here? As far as I recall there was a discussion regarding same element access from different work-items even in case we are writing the same value in it. The discussion was mentioning the article by [Hans-J. Boehm](https://hboehm.info/boehm-hotpar11.pdf.)
I mean, even the following code would be a data race:
```
                if (!std::is_same_v<decltype(ref_1), typename AccT::reference>) {
                  res_acc[0] = false;
                }
```

Reply:
Looks like we can use [`group_barier`](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_group_barrier) here.

```
myQueue.submit([&](handler & cgh) {
  cgh.parallel_for(
      nd_range<3>(range<3>(4, 4, 4), range<3>(2, 2, 2)), [=](nd_item<3> item) {
        //[kernel code]
        // Internal synchronization
        group_barrier(item.get_group());
        //[kernel code]
      });
});
```

Reply:
Changed `res` to bool array with result for each item.